Deprecate gather, gatherN, gatherUnordered in favor of parSequence, parSequenceN, parSequenceUnordered

jozic opened the pull request:
addresses #1130 

- [x] gatherX to parSequenceX
- [x] wanderX to parTraverseX
- [ ] update https://github.com/monix/monix.io

jozic commented:
@Avasil @oleg-py @alexandru 
please check this out, if this looks good, i can continue with the rest

Avasil commented on monix-eval/shared/src/main/scala/monix/eval/Task.scala:
```diff
@@ -3517,36 +3517,46 @@ object Task extends TaskInstancesLevel1 {
 
+  /** Alias for [[parSequence]] */
+  @deprecated("Use parSequence", "3.2.0")
+  def gather[A, M[X] <: Iterable[X]](in: M[Task[A]])(implicit bf: BuildFrom[M[Task[A]], A, M[A]]): Task[M[A]] =
```
We should put deprecated methods to `TaskDeprecated.Companion` to clean normal API

There is also slightly different convention for description with
```
/**
   * DEPRECATED — please use XXX
   */
```

jozic commented:
@Avasil thanks for feedback!
 I've moved deprecated methods to `TaskDeprecated.Companion` and also added renamed classes to the mima filter
I've also tried to revert changes to benchmarks, but current version wasn't compilable due to deprecation errors, so I've disabled `benchmarksPrev`
CI seems happy now

Please check it out again

Avasil commented:
@jozic Looks good, we can add the rest :+1:

I think we can wait with `monix.io` docs update until `3.2.0` is released

jozic commented:
@Avasil I've added second commit with changes for `wander` and friends
lmk if anything need to be changed
